aws/endpoints/endpoints.go

```diff
@@ -21,8 +37,22 @@ type Options struct {
 	// be returned. This endpoint may not be valid. If StrictMatching is
 	// enabled only services that are known to support dualstack will return
 	// dualstack endpoints.
+	//
+	// Deprecated: This option will continue to function for S3 and S3 Control for backwards compatibility.
+	// DualStackEndpoint should be used to enable usage of a service's dual-stack endpoint for all service clients
+	// moving forward. For S3 and S3 Control, when DualStackEndpoint is set to a non-zero value it takes higher
+	// precedence then this option.
 	UseDualStack bool
 
+	// Sets the resolver to resolve the endpoint as a dual-stack endpoint
+	// for the service.
+	//
+	// When enabled, the resolver may in some cases return an endpoint using either the partition or services
+	// dual-stack endpoint pattern which may not be valid or available. In the event that the service or partition
+	// does not have a default dual-stack pattern, and the client's configured region is not an explicitly modeled
+	// endpoint an error will be returned.
+	DualStackEndpoint DualStackEndpoint
```

Comment:
If a test case exists that errors as described in this comment, could you point me to it?

Reply:
Looks like I forgot to port some of the V2 tests where this condition is accounted for:
https://github.com/skmcgrail/aws-sdk-go-v2/blob/dualStackEndpoints/internal/endpoints/endpoints_test.go#L292

Will port over.

Reply:
I stand corrected, the tests are different just because of the general model structure being flattened over in V2. The test in question in v1 are: https://github.com/skmcgrail/aws-sdk-go/blob/dualStackEndpoints/aws/endpoints/v3model_test.go#L311